src/main/java/org/example/springintro/services/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package org.example.springintro.services.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.springintro.dto.shoppingcart.AddToCartRequestDto;
+import org.example.springintro.dto.shoppingcart.CartItemResponseDto;
+import org.example.springintro.dto.shoppingcart.ShoppingCartDto;
+import org.example.springintro.dto.shoppingcart.UpdateCartItemRequestDto;
+import org.example.springintro.mapper.ShoppingCartMapper;
+import org.example.springintro.model.Book;
+import org.example.springintro.model.CartItem;
+import org.example.springintro.model.ShoppingCart;
+import org.example.springintro.model.User;
+import org.example.springintro.repository.book.BookRepository;
+import org.example.springintro.repository.shoppingcart.CartItemRepository;
+import org.example.springintro.repository.shoppingcart.ShoppingCartRepository;
+import org.example.springintro.services.ShoppingCartService;
+import org.example.springintro.services.UserService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final UserService userService;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartDto getCartForCurrentUser(Pageable pageable) {
+        User currentUser = userService.getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(currentUser)
+                .orElseThrow(() -> new RuntimeException("Shopping cart not found"));
+
+        List<CartItemResponseDto> cartItemsDto = cart.getCartItems().stream()
+                .map(shoppingCartMapper::toCartItemDto)
+                .skip((long) pageable.getPageNumber() * pageable.getPageSize())
+                .limit(pageable.getPageSize())
+                .toList();
+
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(cart);
+        shoppingCartDto.setCartItems(cartItemsDto);
+        return shoppingCartDto;
+    }
+
+    @Override
+    public ShoppingCartDto addBookToCart(Long bookId, int quantity) {
+        User currentUser = userService.getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(currentUser)
+                .orElseGet(() -> createNewCartForUser(currentUser));
```

Comment:
you can register shop cart when user registers.

Like:
```
@Override
    public void createShoppingCart(User user) {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.setUser(user);
        repository.save(shoppingCart);
    }
```

and in UserServiceImpl register method 
`cartService.createShoppingCart(user);`